### PR TITLE
be 08 add devise token auth

### PR DESCRIPTION
**Pull request by DanielOldak**
https://trello.com/c/xtZ1nuZT/32-be-08-add-signin-signup-h

### CODE REVIEW

File: db/migrate/20210817103931_devise_token_auth_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      t.boolean  :allow_password_change, :default => false
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      ## Confirmable
+      t.string   :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (PatrykTyszka) [inline db/migrate/20210817103931_devise_token_auth_create_users.rb:24]: It wasn't required, was it?

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (PatrykTyszka) [inline spec/models/user_spec.rb:5]: remove if you don't need it.

File: spec/factories/users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :user do
+    
+  end
+end
```
Comment (PatrykTyszka) [inline spec/factories/users.rb:5]: remove

**Comment by DanielOldak** [comment]
User is able to log in
log in header
log in response

File: config/routes.rb [hunk 4]
```diff
@@ -2,6 +2,7 @@
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   namespace :api do
     namespace :v1 do
+      mount_devise_token_auth_for 'User', at: 'auth'
```
[the remark this answers is not in the document]
Comment (DanielOldak) [reply config/routes.rb:5]: Yes, we have agreed to leave these routes this way

File: Gemfile [hunk 5]
```diff
@@ -24,8 +24,8 @@ gem 'bootsnap', '>= 1.4.4', require: false
 
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
 # gem 'rack-cors'
-
 gem 'active_model_serializers', '~> 0.10'
+gem 'devise_token_auth'
```
Comment (h4b00) [inline Gemfile:28]: Add pesimistic version

**Review by h4b00 — APPROVED** [review]
Apply @PatrykTyszka comments, add pessimistic version and feel free to merge it.
Gj!

File: config/initializers/cors.rb [hunk 6]
```diff
@@ -9,8 +9,9 @@
...
-             headers: :any,
-             methods: %i[get post put patch delete options head],
-             credentials: true
+            headers: :any,
+            methods: %i[get post put patch delete options head],
+            credentials: false,
+            expose: %w[access-token expiry token-type uid client]
   end
-end
+end
```
Comment (PatrykTyszka) [inline config/initializers/cors.rb:17]: empty line at the end of file.

File: db/migrate/20210817103931_devise_token_auth_create_users.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
+class DeviseTokenAuthCreateUsers < ActiveRecord::Migration[6.1]
+  def change
+
+    create_table(:users) do |t|
+      ## Required
+      t.string :provider, :null => false, :default => "email"
```
[the remark this answers is not in the document]
Comment (DanielOldak) [reply db/migrate/20210817103931_devise_token_auth_create_users.rb:6]: Without it i get ActiveRecord::InvalidStatement error in response

File: db/migrate/20210817103931_devise_token_auth_create_users.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      t.string :uid, :null => false, :default => ""
```
[the remark this answers is not in the document]
Comment (DanielOldak) [reply db/migrate/20210817103931_devise_token_auth_create_users.rb:7]: Without it i get ActiveRecord::InvalidStatement error in response

**Review by PatrykTyszka — APPROVED** [review]
Ok we can merge it as is. Please start working on the next task GJ!

**Pull request merged**